crates/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -43,14 +43,75 @@ impl BrilligContext {
 
     /// Allocates an array of size `size` and stores the pointer to the array
     /// in `pointer_register`
-    pub(crate) fn allocate_array(&mut self, pointer_register: RegisterIndex, size: u32) {
-        let array_pointer = self.memory.allocate(size as usize);
+    pub(crate) fn allocate_array(
+        &mut self,
+        pointer_register: RegisterIndex,
+        size: u32,
+        prefilled: bool,
+    ) {
+        let allocation = self.memory.allocate(size as usize);
+
+        // If the array is prefilled (for example, parameter arrays), then we do not need to expand memory
+        if !prefilled {
```

Comment:
I can see how this fixes the bug, though I wonder if this is leaking an abstraction -- ie parameter arrays are not a concept in this part of the code.

Will investigate the ssa ir!